Detect string formatting with number

Is there a rule I need to set?

```bash
echo "print 'asd %s' % 1\nfoo()" > test.py
flake8 test.py
test.py:2:1: F821 undefined name 'foo'
```

Versions:
```
flake8 --version
3.6.0 (flake8_pep3101: 1.2.1, flake8_quotes: 2.1.1, mccabe: 0.6.1, pycodestyle: 2.4.0, pyflakes: 2.0.0) CPython 2.7.15 on Darwin```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -6,7 +6,9 @@ Changelog
 1.2.2 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Warn about `"string %i" % 33` which was previously ignored.
+  Reported by @unformatt.
+  [gforcada]
 
 
 1.2.1 (2018-04-05)

--- a/flake8_pep3101.py
+++ b/flake8_pep3101.py
@@ -24,10 +24,10 @@ class Flake8Pep3101(object):
                 tree = ast.parse(f.read())
 
         for stmt in ast.walk(tree):
-            if isinstance(stmt, ast.BinOp) and \
-                    isinstance(stmt.op, ast.Mod):
-                if isinstance(stmt.left, ast.Num) or \
-                        isinstance(stmt.right, ast.Num):
+            if self._is_module_operation(stmt):
+                if self._is_left_hand_number(stmt):
+                    continue
+                if self._is_modulo_variable_and_number(stmt):
                     continue
 
                 if isinstance(stmt.left, (ast.Str, ast.Name)):
@@ -37,3 +37,18 @@ class Flake8Pep3101(object):
                         self.message,
                         type(self),
                     )
+
+    @staticmethod
+    def _is_module_operation(stmt):
+        return isinstance(stmt, ast.BinOp) and isinstance(stmt.op, ast.Mod)
+
+    @staticmethod
+    def _is_left_hand_number(stmt):
+        """Check if it is a case of `44 % SOMETHING`."""
+        return isinstance(stmt.left, ast.Num)
+
+    @staticmethod
+    def _is_modulo_variable_and_number(stmt):
+        """Check if it is a case of `var % 44`."""
+        return isinstance(stmt.right, ast.Num) and \
+            isinstance(stmt.left, ast.Name)

--- a/run_tests.py
+++ b/run_tests.py
@@ -185,6 +185,19 @@ class TestFlake8Pep3101(unittest.TestCase):
         ret = list(checker.run())
         self.assertEqual(len(ret), 0)
 
+    def test_right_hand_string_left_hand_number(self):
+        file_path = self._given_a_file_in_test_dir('\n'.join([
+            'print("asd %s" % 1)',
+        ]))
+        checker = Flake8Pep3101(None, file_path)
+        ret = list(checker.run())
+        self.assertEqual(len(ret), 1)
 
-if __name__ == '__main__':
-    unittest.main()
+    def test_right_hand_string_left_hand_variable(self):
+        file_path = self._given_a_file_in_test_dir('\n'.join([
+            'a = 44',
+            'print("asd %s" % a)',
+        ]))
+        checker = Flake8Pep3101(None, file_path)
+        ret = list(checker.run())
+        self.assertEqual(len(ret), 1)

